### PR TITLE
Unicode trigger labels

### PR DESCRIPTION
**Pull request by tab-cmd**
# Overview

Update `TriggerHandler` to support reading and writing of unicode trigger labels. This is required for operation of the `RSVP Time Test Calibration`.

## Ticket

[Pivotal Ticket Link](https://www.pivotaltracker.com/story/show/180972841)

## Contributions

- `helpers/triggers.py`: add encoding of `utf-8` to TriggerHandler
- `helpers/tests/test_triggers.py`: update to reflect encoding as a kwarg
- `helpers/README.md`: added a readme with high level information about the submodules. Added detailed trigger information.
- `task/task_registry.py`: encountered odd behavior testing this unicode fix with a mistyped task name. This change fails bci_main sooner with a more helpful message detailing the available registered tasks. 

## Test

- `make test-all`: all tests pass
- `offline_analysis.py`: ensured triggers loaded correctly

## Documentation

- Are documentation updates required? In-line, README, or [documentation](https://github.com/BciPy/bcipy.github.io)? Added a README detailing the trigger module.

### CODE REVIEW

File: bcipy/helpers/README.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+# BciPy Helpers
```
Comment (nik-sm) [inline bcipy/helpers/README.md:1]: Awesome documentation!🔥

File: bcipy/helpers/README.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+# apply a 2 second offset to all timestamps loaded 
+triggers = TriggerHandler.load(path_to_trigger_save_location, offset=2.0)
+```
+
+### Trigger File Format
+
+Triggers as written from BciPy tasks are assumed to have the following structure,
+
+1. A single trigger is written per line with three columns (label type timestamp).
+2. If a clock offset is present, it is written with the type offset and will be added to any timestamp values written in the file. When using various clocks that don't
```
Comment (nik-sm) [inline bcipy/helpers/README.md]: Is it possible to include multiple offsets? For example

```
starting_offset offset -5.0
another_offset offset 5.0
...
```

File: bcipy/helpers/README.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+
+### Loading
+
+To load a BciPy triggers.txt file, the TriggerHandler load method can be used. Because it is a staticmethod, the class does not need to be initialized before the load method is used. 
+
+```python
+from bcipy.helpers.triggers import TriggerHandler
+
+path_to_trigger_save_location = './triggers.txt'
```
Comment (nik-sm) [inline bcipy/helpers/README.md:93]: Is there a reason why `.txt` suffix is included during load, but not during save?

**Review by nik-sm — APPROVED** [review]
Looks good!

File: bcipy/helpers/README.md (hunk 3 above)
Comment (tab-cmd) [reply bcipy/helpers/README.md:93]: No reason. It should be changed though. I have a ticket in the backlog to remove extensions from our other file types (session, raw data) and will add this to it! https://www.pivotaltracker.com/story/show/181115524

File: bcipy/helpers/README.md (hunk 2 above)
Comment (tab-cmd) [reply bcipy/helpers/README.md]: That's a good callout, we only use the first offset written. The other will be loaded as a trigger and can be applied using other functions to the other loaded Triggers if desired. https://github.com/CAMBI-tech/BciPy/blob/1.5.1/bcipy/helpers/triggers.py#L274

File: bcipy/helpers/README.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+## Triggers
+
+Triggers consist of a label, type and a timestamp. These are what is used to align events both in real-time and after a session for further processing or model training. During operation of a BciPy task, Triggers are generated when display or other events of interest occur. These are timestamped using a monotonic clock (See helpers/clock.py).
+
+A new Trigger may defined as follows:
+```python
+from bcipy.helpers.triggers import Trigger, TriggerType
+
+# label can be any utf-8 complaint string
```
Comment (lawhead) [inline bcipy/helpers/README.md]: This should be 'compliant'.

File: bcipy/helpers/triggers.py [hunk 5]
```diff
@@ -212,6 +212,8 @@ class TriggerHandler:
     writing triggers and loading triggers from a txt file.
     """
 
+    encoding = 'utf-8'
```
Comment (lawhead) [inline bcipy/helpers/triggers.py]: We are getting warnings on all of our `open` statements throughout BciPy. Should we have a constant in `system_utils` for this?

**Review by lawhead — APPROVED** [review]

File: bcipy/helpers/triggers.py (hunk 5 above)
Comment (tab-cmd) [reply bcipy/helpers/triggers.py]: I'm running python 3.8.10. I get warnings for other things when I run the tests or suite, but none for open. Can you paste the warning here?

Comment (lawhead) [reply bcipy/helpers/triggers.py]: I'm seeing these as pylint warnings:

   `Using open without explicitly specifying an encoding`

Comment (lawhead) [reply bcipy/helpers/triggers.py]: Thanks for fixing these!

**Pull request merged**